src/modules/ekf2/ekf2_main.cpp

```diff
@@ -792,102 +781,24 @@ void Ekf2::task_main()
-				ctrl_state.x_vel = v_b(0);
-				ctrl_state.y_vel = v_b(1);
-				ctrl_state.z_vel = v_b(2);
-
-
-				// Local Position NED
-				float position[3];
-				_ekf.get_position(position);
-				ctrl_state.x_pos = position[0];
-				ctrl_state.y_pos = position[1];
-				ctrl_state.z_pos = position[2];
-
-				// Attitude quaternion
-				q.copyTo(ctrl_state.q);
-
-				_ekf.get_quat_reset(&ctrl_state.delta_q_reset[0], &ctrl_state.quat_reset_counter);
-
-				// Acceleration data
-				matrix::Vector<float, 3> acceleration(sensors.accelerometer_m_s2);
-
-				float accel_bias[3];
-				_ekf.get_accel_bias(accel_bias);
-				ctrl_state.x_acc = acceleration(0) - accel_bias[0];
-				ctrl_state.y_acc = acceleration(1) - accel_bias[1];
-				ctrl_state.z_acc = acceleration(2) - accel_bias[2];
-
-				// compute lowpass filtered horizontal acceleration
-				acceleration = R_to_body.transpose() * acceleration;
-				_acc_hor_filt = 0.95f * _acc_hor_filt + 0.05f * sqrtf(acceleration(0) * acceleration(0) +
-						acceleration(1) * acceleration(1));
-				ctrl_state.horz_acc_mag = _acc_hor_filt;
-
-				ctrl_state.airspeed_valid = false;
-
-				// use estimated velocity for airspeed estimate
-				if (_airspeed_mode.get() == control_state_s::AIRSPD_MODE_MEAS) {
-					// use measured airspeed
-					if (PX4_ISFINITE(airspeed.indicated_airspeed_m_s) && now - airspeed.timestamp < 1e6
-					    && airspeed.timestamp > 0) {
-						ctrl_state.airspeed = airspeed.indicated_airspeed_m_s;
-						ctrl_state.airspeed_valid = true;
-					}
-
-				} else if (_airspeed_mode.get() == control_state_s::AIRSPD_MODE_EST) {
-					if (_ekf.local_position_is_valid()) {
-						ctrl_state.airspeed = sqrtf(velocity[0] * velocity[0] + velocity[1] * velocity[1] + velocity[2] * velocity[2]);
-						ctrl_state.airspeed_valid = true;
-					}
-
-				} else if (_airspeed_mode.get() == control_state_s::AIRSPD_MODE_DISABLED) {
-					// do nothing, airspeed has been declared as non-valid above, controllers
-					// will handle this assuming always trim airspeed
-				}
-
-				// publish control state data
-				if (_control_state_pub == nullptr) {
-					_control_state_pub = orb_advertise(ORB_ID(control_state), &ctrl_state);
-
-				} else {
-					orb_publish(ORB_ID(control_state), _control_state_pub, &ctrl_state);
-				}
-			}
-
-
 			{
 				// generate vehicle attitude quaternion data
 				struct vehicle_attitude_s att = {};
 				att.timestamp = now;
 
 				q.copyTo(att.q);
+				_ekf.get_quat_reset(&att.delta_q_reset[0], &att.quat_reset_counter);
 
-				att.rollspeed = gyro_rad[0];
-				att.pitchspeed = gyro_rad[1];
-				att.yawspeed = gyro_rad[2];
+				float gyro_bias[3] = {};
+				_ekf.get_gyro_bias(gyro_bias);
+
+				att.rollspeed = sensors.gyro_rad[0] - gyro_bias[0];
+				att.pitchspeed = sensors.gyro_rad[1] - gyro_bias[1];
+				att.yawspeed = sensors.gyro_rad[2] - gyro_bias[2];
+
+				att.roll_rate_bias = gyro_bias[0];
+				att.pitch_rate_bias = gyro_bias[1];
+				att.yaw_rate_bias = gyro_bias[2];
```

Comment:
The only place it is used at the moment is in the mc attitude controller which uses sensor_gyro directly and applies the bias corrections from the EKF and also uses the vehicle_attitude topic.  I expect other controllers with body rate feedback to follow the same approach over time. 

Moving it into  a separate topic makes more sense in the future if and when we split out the rate and attitude controllers.